pynamodb/models.py

```diff
@@ -250,6 +251,16 @@ def __init__(self, hash_key=None, range_key=None, **attributes):
             attributes[self._range_keyname] = range_key
         super(Model, self).__init__(**attributes)
 
+    def get_hash_key(self):
+        key_name = self._hash_keyname
+        return getattr(self, key_name)
+
+    def get_range_key(self):
+        key_name = self._range_keyname
+        if key_name is not None and hasattr(self, key_name):
```

Comment:
```suggestion
        if key_name is not None:
            getattr(self, key_name, None)
```

iirc, `hasattr(self, key_name)` is just a `getattr(self, key_name, None)` that discards the result: https://hynek.me/articles/hasattr/